content/client-server-api/_index.md

```diff
@@ -336,42 +337,92 @@ to prevent the access token being leaked in access/HTTP logs. The query
-one time.
+invalidate any access and refresh tokens previously assigned to that device.
+
+### Refreshing access tokens
+
+{{% added-in v="1.3" %}}
+
+Access tokens can expire after a certain amount of time. Any HTTP
+calls that use an expired access token will return with an error code
+`M_UNKNOWN_TOKEN`, preferably with `soft_logout: true`. When a client
+receives this error and it has a refresh token, it should attempt to
+refresh the access token by calling `/refresh`. Clients can also
+refresh their access token at any time, even if it has not yet
+expired. If the token refresh succeeds, it should use the new token
+for future requests, and can re-try previously-failed requests with
+the new token. When an access token is refreshed, a new refresh token
+may be returned; if a new refresh token is given, the old refresh
+token will be invalidated, and the new refresh token should be used
+when the access token needs to be refreshed.
+
+If the token refresh fails, then the client can treat it as a [soft
+logout](#soft-logout), if the error response included a `soft_logout:
+true` property, and attempt to obtain a new access token by re-logging
+in. If the error response does not include a `soft_logout: true`
+property, the client should consider the user as being logged out.
+
+Handling of clients that do not support refresh tokens is up to the
+homeserver; clients indicate their support for refresh tokens by
+including a `refresh_token: true` property in the request body of the
+`/login` and `/register` endpoints. For example, homeservers may allow
+the use of non-expiring access tokens, or may expire access tokens
+anyways and rely on soft logout behaviour on clients that don't
+support refreshing.
 
 ### Soft logout
 
-When a request fails due to a 401 status code per above, the server can
-include an extra response parameter, `soft_logout`, to indicate if the
-client's persisted information can be retained. This defaults to
-`false`, indicating that the server has destroyed the session. Any
-persisted state held by the client, such as encryption keys and device
-information, must not be reused and must be discarded.
+A client can be in a "soft logout" state if the server requires
+re-authentication before continuing, but does not want to invalidate
+the client's session. That is, any persisted state held by the client,
+such as encryption keys and device information, must not be reused and
+must be discarded, and can be re-used if the client successfully
+re-authenticates.
 
-When `soft_logout` is true, the client can acquire a new access token by
-specifying the device ID it is already using to the login API. In most
-cases a `soft_logout: true` response indicates that the user's session
-has expired on the server-side and the user simply needs to provide
-their credentials again.
+The server indicates that the client is in a soft logout state by
+including a `soft_logout: true` parameter in an `M_UNKNOWN_TOKEN`
```

Comment:
https://spec.matrix.org/unstable/client-server-api/#common-error-codes only defines the `soft_logout` parameter under `M_UNKNOWN_TOKEN`, so it should be OK.  And I kind of want us to prefer using the error codes in the spec rather than the HTTP status codes, since that's what we say clients should prefer.